lib/spack/spack/test/cmd/module.py

```diff
@@ -83,7 +84,7 @@ def test_remove_and_add(database, module_type):
         return
 
     rm_cli_args = ['rm', '-y', 'mpileaks']
-    module_files = _module_files(module_type, 'mpileaks')
+    module_files = _module_files(module_type, 'mpileaks ^mpich')
```

Comment:
why's the extra `^mpich` needed?

Reply:
The extra `^mpich` was needed because at some point the preferred provider for the spec changed to be `intel-parallel-studio`. This changed during a rebase, and another later rebase made it unnecessary - so I just removed it.